src/plugins/workspace/server/routes/index.ts

```diff
@@ -172,6 +173,19 @@ export function registerRoutes({
     },
     router.handleLegacyErrors(async (context, req, res) => {
       const { attributes } = req.body;
+      const rawRequest = ensureRawRequest(req);
+      const authInfo = rawRequest?.auth?.credentials?.authInfo as { user_name?: string } | null;
+      const permissions = Array.isArray(attributes.permissions)
+        ? attributes.permissions
+        : [attributes.permissions];
+
+      if (!!authInfo?.user_name) {
+        permissions.push({
+          type: 'user',
+          userId: authInfo.user_name,
+          modes: [WorkspacePermissionMode.Management],
+        });
```

Comment:
if user have add himself with management permission on UI, will this have duplicate data?

Reply:
never mind, acl will handle duplicate data.